bucket/7zip.json

```
@@ -1,40 +1,107 @@
     "license": {
-        "identifier": "Freeware,LGPL-2.0-only,BSD-3-Clause",
+        "identifier": "Freeware,GNU LGPL,unRAR license,BSD 3-Clause,BSD 2-clause",
         "url": "https://www.7-zip.org/license.txt"
     },
     "architecture": {
         "64bit": {
-            "url": "https://7-zip.org/a/7z2404-x64.msi",
-            "hash": "6a9088743ae698b6c329576769d131a29ee7abb85ac36972fffaf21d8bf8dbaa"
+            "url": [
+                "https://www.7-zip.org/a/7z2301-x64.exe#/7z2301.exe",
+                "https://raw.githubusercontent.com/brave-simpletons/scoop-the-business/main/assets/7zip/install-context.reg",
+                "https://raw.githubusercontent.com/brave-simpletons/scoop-the-business/main/assets/7zip/uninstall-context.reg"
+            ],
+            "hash": [
+                "26cb6e9f56333682122fafe79dbcdfd51e9f47cc7217dccd29ac6fc33b5598cd",
+                "a3e7a9e3088e4316a7281ef7f0e95168dcdf68d28653b6e44db7b61c9eb01ae6",
+                "494fb5be4bc0b4323701319e72d31acb8c1c8a5d7c94f0e26620a415f551fcfa"
+            ]
         },
         "32bit": {
-            "url": "https://7-zip.org/a/7z2404.msi",
-            "hash": "e98109f281de9f48a0fddb2e6367449bd636f9656a29930603a5e68b213ef525"
+            "url": [
+                "https://www.7-zip.org/a/7z2301.exe#/7z2301.exe",
+                "https://raw.githubusercontent.com/brave-simpletons/scoop-the-business/main/assets/7zip/install-context.reg",
+                "https://raw.githubusercontent.com/brave-simpletons/scoop-the-business/main/assets/7zip/uninstall-context.reg"
+            ],
+            "hash": [
+                "9b6682255bed2e415bfa2ef75e7e0888158d1aaf79370defaa2e2a5f2b003a59",
+                "a3e7a9e3088e4316a7281ef7f0e95168dcdf68d28653b6e44db7b61c9eb01ae6",
+                "494fb5be4bc0b4323701319e72d31acb8c1c8a5d7c94f0e26620a415f551fcfa"
+            ]
+        },
+        "arm64": {
+            "url": [
+                "https://www.7-zip.org/a/7z2301-arm64.exe#/7z2301.exe",
+                "https://raw.githubusercontent.com/brave-simpletons/scoop-the-business/main/assets/7zip/install-context.reg",
+                "https://raw.githubusercontent.com/brave-simpletons/scoop-the-business/main/assets/7zip/uninstall-context.reg"
+            ],
+            "hash": [
+                "6fa4cb35cbebb0a46b8bbc22d1686a340e183c1f875d8b714efdc39af93debda",
+                "a3e7a9e3088e4316a7281ef7f0e95168dcdf68d28653b6e44db7b61c9eb01ae6",
+                "494fb5be4bc0b4323701319e72d31acb8c1c8a5d7c94f0e26620a415f551fcfa"
+            ]
         }
     },
-    "extract_dir": "Files\\7-Zip",
-    "bin": "7z.exe",
+    "pre_install": [
+        "Write-Host \"Downloading the temporary extractor...\"",
+        "$7zr = Join-Path $env:TMP '7zr.exe'",
+        "Invoke-WebRequest https://www.7-zip.org/a/7zr.exe -OutFile $7zr",
+        "Get-ChildItem -Path $dir -Filter 7z*.exe | Select-Object -First 1 | ForEach-Object {",
```

Have you tried using `$fname`?
```suggestion
        "Get-ChildItem -Path $dir -Filter $fname | Select-Object -First 1 | ForEach-Object {",
```

---

yep i tried, but the way $fname works, it uses the last one in the list of url... which would be a reg file ... that's why I used that work around... That said, since now any exe name is the same (like 7z2301.exe) I will instead try to simplify and use the real name of the exe and remove this "search" and "foreach"...

---

I change the filename to use 7z.$version.exe instead and using it in the pre_install script to simplify it.